### PR TITLE
Generator

### PR DESCRIPTION
**Pull request by lukasz-smogorzewski**
Added generating boards.

### CODE REVIEW

File: src/generator/boardGenerator.java [hunk 1]
```diff
@@ -0,0 +1,222 @@
+package generator;
+
+import java.util.Random;
+
+public class boardGenerator {
```
Comment (kam193) [inline src/generator/boardGenerator.java]: To unification, please use PascalCase

File: src/generator/boardGenerator.java [hunk 2]
```diff
@@ -0,0 +1,222 @@
...
+    }
+
+    //this generator could generate board with two fields that don't have routes to one another
+
+    private void generate() {
+        Random rand = new Random();
+        int n;
+        for (int i = 0; i != size; ++i) {
+            for (int j = 0; j != size; ++j) {
+                if ((n = rand.nextInt(2)) == 0)
```
Comment (kam193) [inline src/generator/boardGenerator.java:36]: Probably it should working like random int form <1, 2>, yes? If yes, please use something like `rand.nextInt(1)+1`, when you use this if, `n` still can be 0.

If not, what should this line do?

File: src/generator/boardGenerator.java [hunk 3]
```diff
@@ -0,0 +1,222 @@
...
+            isChanged = false;
+            for (int i = 0; i != size; ++i) {
+                for (int j = 0; j != size; ++j) {
+                    if (board[i][j] == -1) {
+                        if ((i == 0 || board[i - 1][j] != -1) && (i == size - 1 || board[i + 1][j] != -1)) {
+                            board[i][j] = 0;
+                            isChanged = true;
+                        }
+                    }
+                    if (board[i][j] == -1) {
```
Comment (kam193) [inline src/generator/boardGenerator.java]: I think, it should be `else` inside the previous statement, not the same if after them

File: src/generator/boardGenerator.java [hunk 4]
```diff
@@ -0,0 +1,222 @@
...
+                            isChanged = true;
+                        }
+                    }
+                }
+            }
+        }
+
+    }
+
+    private boolean checkReq() {
```
Comment (kam193) [inline src/generator/boardGenerator.java]: Maybe can we use better name? This check only if the BLANK fields are on less than 3/4 of space, nothing more, so e.g. `checkBlankFieldsSpaceReq` or something better

File: src/generator/boardGenerator.java [hunk 5]
```diff
@@ -0,0 +1,222 @@
...
+    //this generator could generate board with two fields that don't have routes to one another
+
+    private void generate() {
+        Random rand = new Random();
+        int n;
+        for (int i = 0; i != size; ++i) {
+            for (int j = 0; j != size; ++j) {
+                if ((n = rand.nextInt(2)) == 0)
+                    n = rand.nextInt(2);
+                board[i][j] = 0 - n;
```
Comment (kam193) [inline src/generator/boardGenerator.java]: TODO: use something like final int or enum, now I'm not totally sure what means which value

File: src/generator/boardGenerator.java [hunk 6]
```diff
@@ -0,0 +1,222 @@
...
+        }
+        Random rand = new Random();
+        for (int i = 0; i != size; ++i) {
+            for (int j = 0; j != size; ++j) {
+                if (board[i][j] < 0) {
+                    temp = rand.nextInt(9) + 1;
+                    board[i][j] = temp;
+                    while (checkNumber(i, j, temp)) {
+                        possibilities[temp - 1] = false;
+                        if (!possibilities[0] && !possibilities[1] && !possibilities[2] && !possibilities[3] && !possibilities[4] && !possibilities[5] && !possibilities[6] && !possibilities[7] && !possibilities[8]) {
```
Comment (kam193) [inline src/generator/boardGenerator.java:121]: Use for loop ;)

File: src/generator/boardGenerator.java [hunk 7]
```diff
@@ -0,0 +1,222 @@
...
+                System.out.print(board[i][j] + " ");
+            }
+            System.out.println();
+        }
+        System.out.println();
+
+    }
+
+    public String toString() {
+        String toReturn = (size + 1) + " " + (size + 1) + "\n";
```
Comment (kam193) [inline src/generator/boardGenerator.java]: Use StringBuilder

File: src/sample/Main.java [hunk 8]
```diff
@@ -16,40 +17,42 @@
...
+        if (args.length == 2) {
             inputFile = args[0];
             outputFile = args[1];
         }
 
         DataInput di = new DataInput();
 
-
-        di.ReadBoard(inputFile);
+        di.readString(i.toString());
```
Comment (kam193) [inline src/sample/Main.java]: I think we shouldn't commit changes in Main.java that are used only for help in local debug etc.

**Review by kam193 — CHANGES_REQUESTED** [review]

File: src/generator/boardGenerator.java (hunk 2 above)
Comment (lukasz-smogorzewski) [reply src/generator/boardGenerator.java:36]: nextInt gives us inclusive 0 and exclusive 2, so rand.nextInt(2) gives 0 or 1, when rand.nextInt(1) always gives 0

File: src/generator/boardGenerator.java (hunk 6 above)
Comment (lukasz-smogorzewski) [reply src/generator/boardGenerator.java:121]: why?

**Pull request merged**